### DISCUSSION TITLE
Disabling multi-line expressions?

### DISCUSSION THREAD

**Discussion by blenderskool** [josdejong/mathjs#2402] [Q&A] [answered]
Hi, is there a way to disable the execution of multi-line expressions(that use `;` or `\n` to separate different lines). I couldn't find any option in the docs for this and I don't have much idea about how mathjs is handling this internally.

After disabling multi-line expressions, I'd expect mathjs to execute only the first line of the multi-line expression or just throw a syntax error and not execute any statement.

**Comment by josdejong** [comment, marked as answer]
There is no built-in option, but you can do some pre/post processing to throw an error or something in case the user entered a multi line expression. Some ideas:

- Just execute any expression. Check if the returned result is a `ResultSet`, if so, throw an error
- Parse the expression. If the parsed tree is a `BlockNode`, throw an error. If not, execute the parsed expression.

**Reply by blenderskool** [reply]
Thanks, @josdejong. I also considered using a regular expression to check if `;` or `\n` appear in the expression(also taking into account for quoted strings where the characters are fine).
I'll try the approaches you shared.

**Reply by josdejong** [reply]
You indeed have to be careful with a regex approach: like you say, these characters can be part of a string, and also, `;` is a valid character used in matrices like `[1, 2; 3, 4]`.
